app/src/main/java/fr/neamar/kiss/DataHandler.java

```diff
@@ -431,6 +432,53 @@ public SearchProvider getSearchProvider() {
         return favorites;
     }
 
+    /**
+     * This method is used to set the specific position of an app in the fav array.
+     *
+     * @param context The mainActivity context
+     * @param id the app you want to set the position of
+     * @param position the new position of the fav
+     */
+    public void setFavoritePosition(MainActivity context, String id, int position) {
+        String favApps = PreferenceManager.getDefaultSharedPreferences(this.context).
+                getString("favorite-apps-list", "");
+        List<String> favAppsList = new ArrayList<>(Arrays.asList(favApps.split(";")));
```

Comment:
I think it would be safer to use DataHandler.getFavorites(), because some favorites might exist in the list but not have an associated POJO, which would make all your calculations wrong.

Using getFavorites, you should be able to compare with the Pojo id and get an index that is guaranteed correct

Reply:
Spoke offline, but because im actually getting the favourite by id from the string list in preferences im getting the right position so this isn't a problem.